README.rst

```diff
@@ -24,6 +24,6 @@ The official home of this guide is https://devguide.python.org.
 Compilation
 -----------
 
-For the compilation of the devguide, Python 3.11+ is needed::
+For the compilation of the devguide::
```

Comment:
This noticed this before, and since we're touching it now anyway:

```suggestion
To render the devguide to HTML under ``_build/html``, run::
```

* Use more familiar/appropriate terminology for docs
* Clarify the phrasing
* Tell users where they can actually find the rendered output